src/pynxtools_raman_multiformat/reader.py

```diff
@@ -0,0 +1,242 @@
+                                        "/spectral_filter_TYPE[",
+                                        "/temp_control_TYPE[",
+                                        "/software_TYPE[",
+                                        "/LENS_OPT["
+
+        ]
+        if self.eln_data.get(key) is None:
+            for string in upper_and_lower_mixed_nexus_concepts:
+                key = key.replace(string,"/[")
+            # add only characters, if they are lower case and if they are not "[" or "]"
+            result = ''.join([char for char in key if not (char.isupper() or char in '[]')])
+            result = result.replace("entry","ENTRY[entry]") # CHANGE THIS LATER
+            if self.eln_data.get(result) is not None:
+                return self.eln_data.get(result)
+            else:
+                return "Parsing error?"
+        return self.eln_data.get(key)
+
+    def get_data(self, key: str, path: str) -> Any:
+        """Returns measurement data from the given eln_data entry."""
+        if path.endswith(("x_values", "y_values","x_values_raman")):
+            return self.txt_data.get(f"data/{path}")
+        else:
+            logger.warning(f"No axis name corresponding to the path {path}.")
+
+
+    def post_process(self) -> None:
+        """
+        Do postprocessing after all files and config file are read.
+        """
+
+        def transform_nm_to_wavenumber(self, lambda_laser, lambda_measurement):
+            stokes_raman_shift = -(1e7 / lambda_measurement - 1e7 / lambda_laser)
+            return stokes_raman_shift
+
+        def get_incident_wavelength_from_NXraman(self):
+            substring = "/beam_incident/wavelength"
+
+            # Find matching keys with contain this substring
+            wavelength_keys = [key for key in self.eln_data if substring in key]
+            # Filter the matching keys for the strings, which contain this substring at the end only
+            filtered_list = [string for string in wavelength_keys if string.endswith(substring)]
+            # get the laser wavelength
+            laser_wavelength = self.eln_data.get(filtered_list[0])
+            return laser_wavelength
+
+        laser_wavelength = get_incident_wavelength_from_NXraman(self)
+        x_values_raman = transform_nm_to_wavenumber(self, laser_wavelength, self.txt_data["data/x_values"])
+
+        self.txt_data["data/x_values_raman"] = x_values_raman
+
+
+
+
+READER = RamanReaderMulti
+
+# Use this command in this .py file folder:
+# dataconverter eln_data.yaml Si-wafer-Raman-Spectrum-1.txt  -c config_file.json --reader raman_multi --nxdl NXraman --output output_raman.nxs
+#
+# Remaining Warnings
+# Could not find value for key /ENTRY[entry]/DATA[data]/x_values_raman/@long_name with value @attrs:/ENTRY[entry]/data/longname_x_raman.
+# Tried prefixes: [('@attrs', '/ENTRY[entry]/data/longname_x_raman')].
+# WARNING: Missing attribute: "/ENTRY[entry]/definition/@URL"

```

Comment:
Does the nxs file have this attribute?

Reply:
The error still occurred. Would stick with it for now and take a look later into pynxtools to understand this. Its not critically for now.